File: 11-body_mass_index/task.ex
defmodule Task do

  def bmIndex(count, values) do
    bmIndex(count, values, [])
  end

  defp bmIndex(_, [], result) do
    Enum.reverse(result)
  end

  defp bmIndex(count, [head_value|tail_values], result) do
    {value1, value2} = head_value
    value_median = bm_Index(Kernel.round(value1 / (value2 * value2) * 10) / 10)
    bmIndex(count - 1, tail_values, [value_median|result])
  end
  
  defp bm_Index(a) when a < 18.5, do: "under"
  defp bm_Index(a) when a < 25.0 and a >= 18.5, do: "normal"
  defp bm_Index(a) when a < 30.0 and a >= 25.0, do: "over"
  defp bm_Index(_), do: "fat"

end

File: 11-body_mass_index/task.erl
-module(task).

-export([bmIndex/2]).

bmIndex(Count, Values) ->
  bmIndex(Count, Values, []).

bmIndex(_, [], Result) ->
  lists:reverse(Result);
bmIndex(Count, [{Value1, Value2}|ValuesTail], Result) ->
  BMI = bm_Index(erlang:round(Value1 / (Value2 * Value2) * 10) / 10),
  bmIndex(Count - 1, ValuesTail, [BMI|Result]).

bm_Index(A) when A < 18.5 ->
  <<"under">>;
bm_Index(A) when A < 25.0, A >= 18.5 ->
  <<"normal">>;
bm_Index(A) when A < 30.0, A >= 25.0 ->
  <<"over">>;
bm_Index(_) -> <<"fat">>.
